modules/statemachine/statemachinemodule.cpp

```diff
@@ -116,6 +118,23 @@ bool StateMachineModule::canGoToState(const std::string& state) const {
     return _machine->canTransitionTo(state);
 }
 
+void StateMachineModule::saveToFile(const std::string& filename,
+                                    std::optional<std::string> directory) const
+{
+    if (!_machine) {
+        LWARNING("Attempting to use uninitialized state machine");
```

Comment:
I think this should be an error?   Warning seems more like:  You are doing something strange that might not work.  Error is:  You are doing something strange that will not work